src/EventStore.Core/TransactionLog/Scavenging/Stages/ChunkExecutor.cs

```diff
@@ -40,7 +40,13 @@ public ChunkExecutor(
 		_chunkSize = chunkSize;
 		_unsafeIgnoreHardDeletes = unsafeIgnoreHardDeletes;
 		_cancellationCheckPeriod = cancellationCheckPeriod;
-		_threads = threads;
+
+		if (threads < 1 || threads > TFChunkScavenger.MaxThreadCount) {
+			_logger.Warning("SCAVENGING: Number of threads specified ({SpecifiedThreads}) is too high, clamping to {AdjustedThreads}",
```

Comment:
This is pretty pedantic, but if the provided thread count is 0 or -1, the value would be clamped to 1 instead of `MaxThreadCount`

Perhaps it would be better to move this log line to after `_threads` are set, and simply log a warning that the thread count was clamped if `threads != _threads`?

Reply:
yeah that's much better, thanks!